Makefile

```diff
@@ -46,29 +46,20 @@ else
 OTELCOL_BUILDER=$(shell command -v ocb)
 endif
 
+.PHONY: go
 go:
-ifeq (, $(shell command -v go 2>/dev/null))
-$(error go command not found. Please install golang. https://go.dev/doc/install )
-endif
-
-envsubst:
-ifeq (, $(shell command -v envsubst 2>/dev/null))
-$(error envsubst command not found. Please install gettext. )
-endif
+	@{ \
+		if ! command -v '$(GO)' >/dev/null 2>/dev/null; then \
+			echo >&2 '$(GO) command not found. Please install golang. https://go.dev/doc/install'; \
+			exit 1; \
+		fi \
+	}
 
-yq:
-ifeq (, $(shell command -v yq 2>/dev/null ))
+.PHONY: goreleaser
+goreleaser:
 	@{ \
-	set -e ;\
-	os=$$(uname | tr A-Z a-z) ;\
-	machine=$$(uname -m) ;\
-	[ "$${machine}" != x86 ] || machine=386 ;\
-	[ "$${machine}" != x86_64 ] || machine=amd64 ;\
-	echo "Installing yq ($${os}/$${machine}) at $(YQ_DIR)";\
-	mkdir -p $(YQ_DIR) ;\
-	curl -sLo $(YQ) https://github.com/mikefarah/yq/releases/download/v$(YQ_VERSION)/yq_$${os}_$${machine} ;\
-	chmod +x $(YQ) ;\
+		if ! command -v '$(GORELEASER)' >/dev/null 2>/dev/null; then \
+			echo >&2 '$(GORELEASER) command not found. Please install goreleaser. https://goreleaser.com/install/'; \
+			exit 1; \
+		fi \
```

Comment:
I tried to adapt the existing pattern of the `go:` rule, but ran into an issue. The Makefile is structured so that only targets that use a tool will depend on the tool’s rule, but the `ifeq` and `error` are evaluated unconditionally by make as the Makefile is loaded.

So I ended up rewriting the `go` target and no `goreleaser` target as these shell snippets.